src/ERC20.sol

```diff
@@ -100,9 +98,15 @@ contract ERC20 is Context, IERC20 {
         address sender,
         address recipient,
         uint256 amount
-    ) public virtual override returns (bool) {
+    ) external virtual override returns (bool) {
+        uint256 _allowance = allowance[sender][msg.sender];
+
+        if (_allowance != type(uint256).max) {
```

Comment:
Why? Is there such a standard that if you approve max, it means infinite?

Reply:
Yes, this is a "standard".